source/air/src/ast.rs

```diff
@@ -181,4 +181,5 @@ pub enum CommandX {
     SetOption(Ident, Ident), // set-option option value (no colon on the option)
     Global(Decl),            // global declarations
     CheckValid(Query),       // SMT check-sat (reporting validity rather than satisfiability)
+    SingularCheckValid(Vec<String>, Vec<Expr>, Vec<Expr>, Span), // check integer-ring proof using Singular: SingularCheckValid{params, ensures, requires, span}
```

Comment:
I'd rather we didn't special-case on singular here.
* Why cannot this carry a `Query`? Can we define a `Query` type for singular, if there are differences?
* Depending on that we may want a separate variant (for solvers with a different kind of Query) or a `Solver` field on `CheckValid`.

Reply:
I can make it fit into `Query` and make an additional boolean field(`is_singular`) in `QueryX`.